app/src/hooks/useGraphFpmmTransactionsFromQuestion.tsx

```diff
@@ -0,0 +1,162 @@
+import { useQuery } from '@apollo/react-hooks'
+import { BigNumber, formatUnits } from 'ethers/utils'
+import gql from 'graphql-tag'
+import { useEffect, useState } from 'react'
+
+import { Status } from '../util/types'
+
+const fragment = gql`
+  fragment TransactionFields on FpmmTransaction {
+    id
+    user {
+      id
+    }
+    fpmm {
+      collateralToken
+    }
+    fpmmType
+    transactionType
+    collateralTokenAmount
+    sharesOrPoolTokenAmount
+    creationTimestamp
+    transactionHash
+  }
+`
+const withFpmmType = gql`
+  query fpmmTransactions($id: ID!, $pageSize: Int, $pageIndex: Int, $fpmmType: String) {
+    fpmmTransactions(
+      where: { fpmm: $id, fpmmType: $fpmmType }
+      first: $pageSize
+      skip: $pageIndex
+      orderBy: creationTimestamp
+      orderDirection: desc
+    ) {
+      ...TransactionFields
+    }
+  }
+  ${fragment}
+`
+const withoutFpmmType = gql`
+  query fpmmTransactions($id: ID!, $pageSize: Int, $pageIndex: Int) {
+    fpmmTransactions(
+      where: { fpmm: $id }
+      first: $pageSize
+      skip: $pageIndex
+      orderBy: creationTimestamp
+      orderDirection: desc
+    ) {
+      ...TransactionFields
+    }
+  }
+  ${fragment}
+`
+export type FpmmTradeDataType = {
+  id: string
+  transactionType: string
+  user: {
+    id: string
+  }
+  fpmm: {
+    collateralToken: string
+  }
+  collateralTokenAddress: string
+  sharesOrPoolTokenAmount: BigNumber
+  collateralTokenAmount: BigNumber
+  creationTimestamp: string
+  transactionHash: string
+  fpmmType: string
+  decimals?: string
+}
+interface FpmmTradeData {
+  id: string
+  transactionType: string
+  user: {
+    id: string
+  }
+  fpmm: {
+    collateralToken: string
+  }
+  collateralTokenAddress: string
+  sharesOrPoolTokenAmount: BigNumber
+  collateralTokenAmount: BigNumber
+  creationTimestamp: string
+  transactionHash: string
+  fpmmType: string
+  decimals?: string
+}
+
+interface Result {
+  fpmmTrade: FpmmTradeData[] | null
+  status: string
+  paginationNext: boolean
+  refetch: any
+}
+const wrangleResponse = (data: any, decimals: number) => {
+  return data.map((trade: FpmmTradeData) => {
+    return {
+      id: trade.id,
+      transactionType:
+        trade.transactionType === 'Add'
+          ? 'Deposit'
+          : trade.transactionType === 'Remove'
+          ? 'Withdraw'
+          : trade.transactionType,
+      user: {
+        id: trade.user.id,
+      },
+      fpmmType: trade.fpmmType,
+      decimals: decimals,
+      collateralTokenAddress: trade.fpmm.collateralToken,
+      sharesOrPoolTokenAmount: parseFloat(formatUnits(trade.sharesOrPoolTokenAmount, decimals)).toFixed(3),
+      creationTimestamp: 1000 * parseInt(trade.creationTimestamp),
+      collateralTokenAmount: trade.collateralTokenAmount,
```

Comment:
I fixed this although i didn't BigNumberfy them since when i include BigNumber types with typescript it does that automatically. Is there other reason to do it?

Reply:
When they're pulled from the subgraph I think they should be strings, strange... Do they work normally as `BigNumber`s?

Reply:
yeah they work great i even have to convert them when using so all is good!